### ISSUE TITLE
@typescript-eslint/require-await - don't throw an error when the method *has* to be async

### ISSUE THREAD

**Issue by kuba-orlik** [typescript-eslint/typescript-eslint#2891]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```js
module.exports = {
	env: { node: true },
	parser: "@typescript-eslint/parser",
	plugins: ["@typescript-eslint"],
	extends: [
		"eslint:recommended",
		"plugin:@typescript-eslint/recommended",
		"plugin:@typescript-eslint/recommended-requiring-type-checking",
	],
	parserOptions: {
		sourceType: "module",
		ecmaFeatures: {
			modules: true,
		},
		project: "./tsconfig.json",
	},
	settings: { jsdoc: { mode: "typescript" } },
};
```

```TS
abstract class Something {
	abstract async doSomething(): Promise<number>;
}

export class ConcreteSomething extends Something {
	async doSomething() {
		return 0;
	}
}
```

```
{
	"compilerOptions": {
		"module": "commonjs",
		"moduleResolution": "node",
		"noImplicitAny": true,
		"noImplicitThis": true,
		"strictNullChecks": true,
		"target": "ES2019",
		"declaration": true,
		"esModuleInterop": true,
		"lib": ["es6", "esnext"],
		"outDir": "lib",
		"checkJs": true,
		"allowJs": true,
		"declarationDir": "@types",
		"resolveJsonModule": true,
		"sourceMap": true
	},
	"include": ["src/**/*"]
}
```

**Expected Result**

I expected eslint not to highlight an error for the `doSomething` method inside the `ConcreteSomething` class, as although it does not have `await`, it has to be async in order to return a Promise, as the abstract method's signature dictates.

**Actual Result**

An error is higlighted at line 6:

```
Async method 'doSomething' has no 'await' expression. [@typescript-eslint/require-await]
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.6.0` |
| `@typescript-eslint/parser`        | `3.6.0 |
| `TypeScript`                       | `4.0.3` |
| `ESLint`                           | `7.4.0` |
| `node`                             | `15.3.0` |

**Comment by bradzacher** [comment]
Your code sample is semantically invalid:

> 'async' modifier cannot be used with 'abstract' modifier. (ts1243)

----

That aside - there's no such thing as a "method that *has* to be `async`".

There's methods that are required to return a promise - but returning a `Promise` is not the same as `async`!
`async` functions return `Promise`s, but functions that return `Promise`s don't have to be marked `async`.

`async` is just a keyword which automatically wraps the return value in a promise if it's not already.

Your example can be written without the `async` keyword just fine:

```ts
abstract class Something {
	abstract doSomething(): Promise<number>;
}

export class ConcreteSomething extends Something {
	doSomething() {
		return Promise.resolve(0);
	}
}
```
